nodeEditor/src/diagram/graphCanvas.tsx

```diff
@@ -841,23 +841,30 @@ export class GraphCanvasComponent extends React.Component<IGraphCanvasComponentP
         this.props.globalState.onRebuildRequiredObservable.notifyObservers();
     }
 
-    processEditorData(editorData: IEditorData) {
-        const frames = this._frames.splice(0);
-        for (var frame of frames) {
-            frame.dispose();
-        }
-
-        this._frames = [];
+    processEditorData(editorData: IEditorData, isImportingAFrame = false) {
+        if (!isImportingAFrame) {
+            const frames = this._frames.splice(0);
+            for (var frame of frames) {
+                frame.dispose();
+            }
 
-        this.x = editorData.x || 0;
-        this.y = editorData.y || 0;
-        this.zoom = editorData.zoom || 1;
+            this._frames = [];
+            this.x = editorData.x || 0;
+            this.y = editorData.y || 0;
+            this.zoom = editorData.zoom || 1;
+        }
 
         // Frames
         if (editorData.frames) {
-            for (var frameData of editorData.frames) {
-                var frame = GraphFrame.Parse(frameData, this, editorData.map);
+            if (isImportingAFrame) {
+                const frame = GraphFrame.Parse(editorData.frames[editorData.frames.length - 1], this, editorData.map);

```

Comment:
we can assume it is editorData.frames[0] right?

Reply:
or even better do we really need to test isImportingAFrame?

Reply:
I refactored this to be more clear after talking with @sebavan .

Because I am appending new frames to editorData.frames, when I import multiple frames in a row, the new frame is not editorData.frames[0], it is at the end of the array.

Reply:
That does not seem to be the case in your code now. I see a call to it once only. Where does the import multiple frames in a row happening?

Reply:
If a user imports a frame, GraphEditor.addFrameFromSerialization() will concat that frame to editorData.frames. If the user imports a second frame in the same session, the second frame is concat-ed to editorData.frame. So the frame I need to parse and add is at the end of editorData.frame.

Reply:
Ok maybe I do not understand it correctly.: if an user imports a frame the editorData of the file will only have one frame in it right?